connector/src/main/java/org/geysermc/connector/entity/PlayerEntity.java

```diff
@@ -221,6 +238,10 @@ public void updateRotation(GeyserSession session, float yaw, float pitch, boolea
     @Override
     public void setPosition(Vector3f position) {
         this.position = position.add(0, entityType.getOffset(), 0);
+        // If this is the player logged in through this Geyser session
+        if (geyserId == 1 && session != null) {
```

Comment:
Why do we check for null here and nowhere else? Because we check for `geyserId`, `session` should never be null.

Reply:
It's null when the position is set for the first time (I think by the constructor of Entity)